hook!-full-game/src/objects/Checkpoint/Checkpoint.gd

```diff
@@ -0,0 +1,24 @@
+extends Area2D
+
+signal reached(checkpoint)
+
+export var id := ""
```

Comment:
All the checkpoint's logic, including getting the data to save and loading save data should be here. You could have a `discovered` bool property to tell if the player already came to this checkpoint.

Don't implement save and load yet just for the checkpoint, we should have a general save system, that's beyond the scope of this task.